lib/cache/sqlite-cache-store.js

```diff
@@ -0,0 +1,399 @@
+      },
+      final (callback) {
+        if (body === null) {
+          return callback()
+        }
+
+        /**
+         * @type {SqliteStoreValue | undefined}
+         */
+        const existingValue = findValue(key, true)
+        if (existingValue) {
+          // Updating an existing response, let's delete it
+          updateValueQuery.run(
+            JSON.stringify(stringifyBufferArray(body)),
+            value.deleteAt,
+            value.statusCode,
+            value.statusMessage,
+            JSON.stringify(stringifyBufferArray(value.rawHeaders)),
+            value.cachedAt,
+            value.staleAt,
+            value.deleteAt,
+            existingValue.id
+          )
+        } else {
+          // New response, let's insert it
+          insertValueQuery.run(
+            url,
+            key.method,
+            JSON.stringify(stringifyBufferArray(body)),
+            value.deleteAt,
+            value.statusCode,
+            value.statusMessage,
+            JSON.stringify(stringifyBufferArray(value.rawHeaders)),
+            value.vary ? JSON.stringify(value.vary) : null,
+            value.cachedAt,
+            value.staleAt,
+            value.deleteAt
+          )
+        }
+
+        callback()
+      }
+    })
+
+    return writable
+  }
+
+  /**
+   * @param {import('../../types/cache-interceptor.d.ts').default.CacheKey} key
+   */
+  delete (key) {
+    if (typeof key !== 'object') {
+      throw new TypeError(`expected key to be object, got ${typeof key}`)
+    }
+
+    this.#deleteByUrlQuery.run(this.#makeValueUrl(key))
+  }
+
+  /**
+   * @param {import('../../types/cache-interceptor.d.ts').default.CacheKey} key
+   * @returns {string}
+   */
+  #makeValueUrl (key) {
+    return `${key.origin}/${key.path}`
+  }
+
+  /**
+   * @param {import('../../types/cache-interceptor.d.ts').default.CacheKey} key
+   * @param {boolean} [canBeExpired=false]
+   * @returns {(SqliteStoreValue & { vary?: Record<string, string[]> }) | undefined}
+   */
+  #findValue (key, canBeExpired = false) {
+    const url = this.#makeValueUrl(key)
+
+    /**
+     * @type {SqliteStoreValue[]}
+     */
+    const values = this.#getValuesQuery.all(url, key.method)
+
+    if (values.length === 0) {
+      // No responses, let's just return early
+      return undefined
+    }
+
+    const now = nowAbsolute()
+    for (const value of values) {
+      if (now >= value.deleteAt && !canBeExpired) {
+        this.#deleteExpiredValuesQuery.run(now)
+        return undefined
+      }
+
+      let matches = true
+
+      if (value.vary) {
+        if (!key.headers) {
+          // Request doesn't have headers so it can't fulfill the vary
+          //  requirements no matter what, let's return early
+          return undefined
+        }
+
+        try {
+          value.vary = JSON.parse(value.vary)
+        } catch (err) {
+          if (this.#errorCallback !== undefined) {
+            this.#errorCallback(err)
+          }
```

Comment:
IMHO If the error callback is just for this I would rather skip it. Would even argue we should crash here if this happens (it shouldn't). @mcollina?